mozilla-try-scripts/gecko-try.sh

```diff
@@ -0,0 +1,144 @@
+done
+
+# Bug 1460178 - sanity check binary downloads of generic-worker before publishing to tooltool...
+
+if ! file "generic-worker-windows-386-v${NEW_VERSION}.exe" | grep -F 'Intel 80386' | grep -F 'for MS Windows'; then
+  echo "Downloaded file doesn't appear to be 386 Windows executable:" >&2
+  file "generic-worker-windows-386-v${NEW_VERSION}.exe" >&2
+  exit 69
+fi
+
+if ! file "generic-worker-windows-amd64-v${NEW_VERSION}.exe" | grep -F 'x86-64' | grep -F 'for MS Windows'; then
+  echo "Downloaded file doesn't appear to be amd64 Windows executable:" >&2
+  file "generic-worker-windows-amd64-v${NEW_VERSION}.exe" >&2
+  exit 70
+fi
+
+cat manifest.tt
+lib/tooltool.py upload -v --authentication-file="$(echo ~/.tooltool-upload)" --message "Upgrade *STAGING* worker types to use generic-worker ${NEW_VERSION}"
+
+git clone git@github.com:mozilla-releng/OpenCloudConfig.git
+cd OpenCloudConfig/userdata/Manifest
+for MANIFEST in *-b.json *-cu.json *-beta.json; do
+  cat "${MANIFEST}" > "${MANIFEST}.bak"
+  cat "${MANIFEST}.bak" | sed "s_\\(generic-worker/releases/download/v\\)[^/]*\\(/generic-worker-windows-\\)_\\1${NEW_VERSION}\\2_" | sed "s_\\(\"generic-worker \\)[^\"]*\\(\"\\)_\\1${NEW_VERSION}\\2_" > "${MANIFEST}"
+  cat "${MANIFEST}" > "${MANIFEST}.bak"
+  THIS_ARCH="$(cat "${MANIFEST}" | sed -n 's/.*\/generic-worker-windows-\(.*\)\.exe.*/\1/p' | sort -u)"
+  if [ "${ARCH}" != "386" ] && [ "${ARCH}" != "amd64" ]; then
+    echo "NOOOOOOO - cannot recognise ARCH" >&2
+    exit 67
+  fi
+  SHA512="$(jq --arg filename "generic-worker-windows-${THIS_ARCH}-v${NEW_VERSION}.exe" '.[] | select(.filename == $filename) .digest' ../../../manifest.tt | sed 's/"//g')"
+  if [ ${#SHA512} != 128 ]; then
+    echo "NOOOOOOO - SHA512 is not 128 bytes: '${SHA512}'" >&2
+    exit 68
+  fi
+  jq --arg sha512 "${SHA512}" --arg componentName GenericWorkerDownload '(.Components[] | select(.ComponentName == $componentName) | .sha512) |= $sha512' "${MANIFEST}.bak" > "${MANIFEST}"
+  rm "${MANIFEST}.bak"
+done
+DEPLOY="deploy: $(git status --porcelain | sed -n 's/^ M userdata\/Manifest\/\(.*\)\.json$/\1/p' | tr '\n' ' ')"
+git add .
+git commit -m "Testing generic-worker ${NEW_VERSION} on *STAGING*
+
+This change does _not_ affect any production workers. Commit made with:
+
+    ./upgrade-gw-betas-cu.sh ${NEW_VERSION}
+
+See https://github.com/petemoore/myscrapbook/blob/master/upgrade-gw-betas-cu.sh" -m "${DEPLOY}"
+OCC_COMMIT="$(git rev-parse HEAD)"
+git push
+
+open_browser_page 'https://github.com/mozilla-releng/OpenCloudConfig/commits/master'
+
+if ! [ -d ~/.mozilla-central ]; then
+  hg clone https://hg.mozilla.org/mozilla-central ~/.mozilla-central
+fi
+
+cd ~/.mozilla-central
+hg up -C
+hg purge
+
+# wait for OCC deployment to complete
+go run "${THIS_SCRIPT_DIR}/waitforOCC.go" "${OCC_COMMIT}"
+
+hg pull -u -r default
+curl -L 'https://bug1400012.bmoattachments.org/attachment.cgi?id=8948627' | hg import -
```

Comment:
Can we check this in to the repo directly instead of curling it?

Reply:
Good point... I'll do that!